src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -796,6 +850,7 @@ public boolean process(Person person, long time) {
       if (type != EncounterType.WELLNESS) {
         person.record.encounterEnd(time, type);
       }
+      materializeCodeField("dischargeDisposition", person.seed);
```

Comment:
I'm guessing the use of reflection for all of these is because the `clone` method isn't doing a deep copy of all the various subfields, and so if you change the Code object in one place it gets propagated throughout every copy of that state? 

Does the use of reflection add any noticeable runtime performance degradation? I've never measured a real-world use case but I've always heard reflection is slow.

Reply:
Yes, I agree that ideally we wouldn't need reflection.

The challenge I encountered was where to put the hooks to "materialize" the ValueSet based codes.

Initially, I thought about creating a custom serializer for Gson. The problem with this is that we need some sort of seed to ensure reproducibility, and the Person seed only gets passed in when the State is processed.

The reflection is really only there to reduce the duplicated code that would be needed within the State subclasses, with the way it is currently "manually" hooked in within the processing of each individual state.

I am sure there is a better way of doing this, I am just not sure what it is and feel that it might need more drastic changes to the way everything currently hangs together.